azure/notebooks/Azure-MNMG-XGBoost.ipynb

```diff
@@ -0,0 +1,1261 @@
+    "\n",
+    "In this example, we are creating a single custom VM image that will be accessible by the user only. We will use a Ubuntu Server 18.04 base image and customize it.\n",
+    "\n",
+    "If you are curious about what else you can configure, take a look at all the available [Azure build parameters for `packer`](https://www.packer.io/docs/builders/azure/arm).\n",
+    "\n",
+    "**Note:** Our resource group already exists in this example. Hence we simply pass in our resource group name in the required parameters `managed_image_resource_group_name` and `build_resource_group_name`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "packer_config = {\n",
+    "  \"builders\": [{\n",
+    "    \"type\": \"azure-arm\", \n",
+    "    \"use_azure_cli_auth\": True,\n",
+    "    \"managed_image_resource_group_name\": resource_group,\n",
+    "    \"managed_image_name\": <the name of the customized VM image>,\n",
+    "    \"custom_data_file\": \"./configs/cloud_init.yaml.j2\",\n",
+    "    \"os_type\": \"Linux\",\n",
+    "    \"image_publisher\": \"Canonical\", \n",
+    "    \"image_offer\": \"UbuntuServer\",\n",
+    "    \"image_sku\": \"18.04-LTS\",\n",
+    "    \"azure_tags\": {\n",
+    "        \"dept\": \"RAPIDS-CSP\",\n",
+    "        \"task\": \"RAPIDS Custom Image deployment\"\n",
+    "    },\n",
+    "    \"build_resource_group_name\": resource_group,\n",
+    "    \"vm_size\": vm_size\n",
+    "  }],\n",
+    "  \"provisioners\": [{\n",
+    "    \"inline\": [\n",
+    "      \"echo 'Waiting for cloud-init'; while [ ! -f /var/lib/cloud/instance/boot-finished ]; do sleep 1; done; echo 'Done'\",\n",
+    "    ],\n",
+    "    \"type\": \"shell\"\n",
+    "  }]\n",
+    "}\n",
+    "\n",
+    "with open(\"packer_config.json\", \"w\") as fh:\n",
+    "    fh.write(json.dumps(packer_config))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### e. Run `packer` build and create the VM image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Uncomment the following line and run to create the custom image\n",
+    "# ! packer build packer_config.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This will take around 15 minutes. Grab a coffee or watch an episode of your favourite tv show and come back. But remember, you will only have to do this once, unless you want to update the packages in the VM image. This means that you can make this custom image once, and then keep on using it for hundreds of times. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### While packer is building the image, you will see an output similar to what is shown below. \n",
```

Comment:
There isn't any output here?

Reply:
The output is in a collapsible html section. For some reason Github Markdown does not render collapsible sections. It will be visible if you open the notebook in Jupyter.

Reply:
Can you add a screenshot then? We'd want it to be readable on GH as well. Everything else looks good

Reply:
I'll attach a screenshot